internal/datasource/group.go

```diff
@@ -0,0 +1,96 @@
+package datasource
+
+import (
+	"context"
+	"fmt"
+
+	"terraform-provider-plural/internal/client"
+	"terraform-provider-plural/internal/common"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	console "github.com/pluralsh/console-client-go"
+)
+
+func NewGroupDataSource() datasource.DataSource {
+	return &groupDataSource{}
+}
+
+type groupDataSource struct {
+	client *client.Client
+}
+
+func (d *groupDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+func (d *groupDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "A representation of a group to organize authorization in your plural console.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "Internal identifier of this group.",
+				Validators:          []validator.String{stringvalidator.ExactlyOneOf(path.MatchRoot("id"))},
```

Comment:
It looks like this can be replaced with Required: true.

Reply:
But we'd want to be able to define a datasource like:

```tf
data "plural_group" {
    name = "developers"
}
```

Wouldn't that impede it?

Reply:
Oh, then I think validator can be skipped.